src/providers/pkcs11/generate_random.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::utils::to_response_status;
+use super::Provider;
+use parsec_interface::operations::psa_generate_random;
+use parsec_interface::requests::{ResponseStatus, Result};
+use std::convert::TryFrom;
+
+impl Provider {
+    pub(super) fn psa_generate_random_internal(
+        &self,
+        op: psa_generate_random::Operation,
+    ) -> Result<psa_generate_random::Result> {
+        let length = u32::try_from(op.size).or_else(|_| {
+            let error = ResponseStatus::ResponseTooLarge;
```

Comment:
I don't think this is the appropriate response status here - `ResponseTooLarge` is about the size of the encoded response payload being larger than the limit set by the admin (see [reference doc](https://parallaxsecond.github.io/parsec-book/parsec_client/status_codes.html)). This should probably be `PsaGenericError`, looking at the other error codes defined by the PSA spec I can't really associate either of them with this case.

Reply:
Ah, I see something similar is happening in the Mbed Crypto provider, just checking it against the `GlobalConfig::buffer_size_limit()` - I'm wondering now if we should implement that check at `BackendHandler` level, because the `BackendHandler` knows what the actual size of the reponse payload is, and should send back `ResponseTooLarge` when the size is exceeded. I'll raise an issue for this.

Reply:
Will update it with PsaGenericError return value.